data_registry/process_manager/task/coverage.py

```diff
@@ -0,0 +1,70 @@
+import gzip
+import shutil
+import tempfile
+from pathlib import Path
+
+import ocdscardinal
+
+from data_registry import models
+from data_registry.process_manager.util import TaskManager, skip_if_not_started
+from exporter.util import Export
+
+
+def get_keys_for_building_block(coverage, building_block):
+    return [key for key in coverage if key.endswith(building_block)]
+
+
+class Coverage(TaskManager):
+    final_output = False
+    json_file_name = "full.jsonl"
+
+    def get_export(self):
+        return Export(self.job.pk, f"{self.json_file_name}.gz")
```

Comment:
The existing logic uses Export's `basename` argument to indicate a file that does not exist until the task is complete. So, we are bending the logic here a bit. I suggest instead assigning a different suffix (e.g. `.coverage`) to avoid potential confusion (we also don't want any weird errors around the exporter task doing something differently because this lockfile exists, or vice versa).

`json_file_path = export.path ` will need to create a different Export instance if the above changes are made.

Reply:
I wasn't sure about using the Export lock file to control if the task is complete. Maybe we could use another totally different approach to control the status. Maybe adding something else to the job context, perhaps?

Reply:
We already have one lock mechanism – unless the new mechanism is much simpler, then we might as well use the same mechanism (or change everything over to a new mechanism).